### ISSUE TITLE
Add route announcer

### ISSUE THREAD

**Issue by arctic-hen7** [arctic-hen7/perseus#124]
This issue is requesting an enhancement to Perseus. Details of the scope will be available in issue labels.
The user described the problem related to this request as follows:
> Right now, Perseus route changes aren't great for accessibility at all. The reason for this is that screen readers don't know that a page change has occurred, and users dependent on these technologies would have no idea that such a change had actually occurred.

The user described the issue as follows:
> This can be remedied quite simply with a route announcer (similar to the one used by NextJS), which would be a Sycamore component with the appropriate ARIA settings at the root of the app (automatically inserted next to `__perseus_content`). Then, this could listen for changes to the router state and announce the title of each new page once it loads. Obviously, we shouldn't do this for the first page, because screen readers will get that one right. The way the NextJS system does this is announcing the document title, falling back to the first `h1` and then falling back to the page path, which should be trivial with Perseus' router infrastructure. This would all be done once the page loads, and title/`h1` extraction would be done with `web_sys`. Notably, some apps will use their own behavior for this functionality, so this will be able to be disabled with a setting in `define_app!`. (As a side note, I do want to keep adding config options there, because it can then act as a single source of truth, rather than some custom, and difficult to debug, config file.)
- The author is willing to attempt an implementation: true 

<details>
<summary>Tribble internal data</summary>

dHJpYmJsZS1yZXBvcnRlZCxDLWVuaGFuY2VtZW50LGF1dGhvci13aWxsaW5nLXRvLWltcGw=

</details>

**Comment by arctic-hen7** [comment]
[Here](https://github.com/vercel/next.js/blob/canary/packages/next/client/route-announcer.tsx)'s the NextJS route announcer for reference (I think this is a pretty good standard to follow in this case).
